scripts/lib-franklin.js

```diff
@@ -141,32 +141,42 @@ export async function decorateIcons(element) {
   // Download all new icons
   const icons = [...element.querySelectorAll('span.icon')];
   await Promise.all(icons.map(async (span) => {
-    const iconName = Array.from(span.classList).find((c) => c.startsWith('icon-')).substring(5);
+    if (span.classList.length < 2 || !span.classList[1].startsWith('icon-')) {
+      return;
+    }
+
+    const iconName = span.classList[1].substring(5);
     if (!ICONS_CACHE[iconName]) {
       ICONS_CACHE[iconName] = true;
       try {
-        const response = await fetch(`${window.hlx.codeBasePath}/icons/${iconName}.svg`);
-        if (!response.ok) {
-          ICONS_CACHE[iconName] = false;
-          return;
-        }
-        // Styled icons don't play nice with the sprite approach because of shadow dom isolation
-        const svg = await response.text();
-        if (svg.match(/(<style | class=)/)) {
-          ICONS_CACHE[iconName] = { styled: true, html: svg };
+        const response = await fetch(`${window.hlx.codeBasePath}${window.hlx.codeBasePath}/icons/${iconName}.svg`);
+        const svgSource = await response.text();
+        if (svgSource.match(/(<style | class=)/)) {
+          ICONS_CACHE[iconName] = { styled: true, html: svgSource };
         } else {
+          const parser = new DOMParser();
+          const parsedSvg = parser.parseFromString(svgSource, 'image/svg+xml');
+          const newSvg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
```

Comment:
Shouldn't this rather be a `symbol` element instead of an `svg` one? The point was to create a reference template for the `<use>` tag below

Reply:
Good point, I will fix it accordingly. thanks